### PR TITLE
Fix Nvidia display driver packages

### LINKED ISSUE #45: nvidia-display-driver-dch-np: checkver still error
## Bug Report

**Package Name:** nvidia-display-driver-dch-np
### Current Behaviour

version 471.41

### Expected Behaviour

should be larger than 500

### Possible Solution

The ```checkver``` is still wrong after May update. The DCH driver is designed for windows 10/11 so should not stay with win7's osid in url

### System details

**Windows version:** [e.g. 7, 8, 10]

**OS architecture:** [e.g. 32bit, 64bit]

**PowerShell version:** [output of `"$($PSVersionTable.PSVersion)"`]

**Additional software:** [(optional) e.g. ConEmu, Git]

#### Scoop Configuration

```json
//# Your configuration here
```

### PR DESCRIPTION
**Pull request by scowalt**
- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

Resolves #45 

I've been running these auto-update instructions in my scoop bucket for a bit. https://github.com/scowalt/scoop-apps/blob/main/bucket/geforce-game-ready-driver.json

I updated the path to reflect the `dl.7z` logic.

- [x] Test `bucket\nvidia-display-driver-np.json` installation
- [x] Test `bucket\nvidia-display-driver-dch-np.json` installation

### CODE REVIEW

File: bucket/nvidia-display-driver-np.json [hunk 1]
```diff
@@ -23,8 +23,8 @@
     ],
     "architecture": {
         "64bit": {
-            "url": "https://us.download.nvidia.com/Windows/472.12/472.12-desktop-win10-win11-64bit-international-whql.exe#/dl.7z_",
```
Comment (scowalt) [inline bucket/nvidia-display-driver-np.json]: This needs to end with `7z_`

**Comment by rasa** [comment]
```batch
--> Running: scoop install  nvidia-display-driver-np.json

Installing 'nvidia-display-driver-np' (528.24) [64bit]
528.24-desktop-win10-win11-64bit-international-dch-whql.exe (813.4 MB) [==========================================] 100%
Checking hash of 528.24-desktop-win10-win11-64bit-international-dch-whql.exe ... ok.
Running installer script...
Linking ~\scoop\apps\nvidia-display-driver-np\current => ~\scoop\apps\nvidia-display-driver-np\528.24
'nvidia-display-driver-np' (528.24) was installed successfully!
Notes
-----
This is only an installer for the NVIDIA display driver. Running `scoop uninstall nvidia-display-driver-np` will only
unregister it from Scoop.
The NVIDIA display driver installer installs and enables the NVIDIA Display Container LS service, which is required for the
NVIDIA Control Panel application but also contains telemetry components.
To disable this functionality and prevent it from being re-enabled on future updates, disable the service using the
following commands:
$ Stop-Service NVDisplay.ContainerLocalSystem
$ Set-Service NVDisplay.ContainerLocalSystem -StartupType Disabled

--> Refreshing environment variables

--> Comparing ARP entries

PS C:\Users\WDAGUtilityAccount\Desktop\bucket>
```

File: bucket/nvidia-display-driver-dch-np.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-    "version": "471.41",
+    "version": "528.24",
     "description": "NVIDIA display driver.",
```
Comment (rasa) [inline bucket/nvidia-display-driver-dch-np.json]: Can we add a suffix that describes what each file is for?  So
```json
"description": "NVIDIA display driver (Declarative Componentized Hardware version).",
```
and I guess add this to the other file:
```json
"description": "NVIDIA display driver (Standard version).",
```

Comment (scowalt) [reply bucket/nvidia-display-driver-dch-np.json]: SGTM. Also, my script has both packages using the same version! I'll make a new comment for that

File: bucket/nvidia-display-driver-np.json [hunk 3]
```diff
@@ -23,8 +23,8 @@
...
-            "hash": "95d8dc470e8f1352b4dbc3b38aefeb5cbbba024826a0541a9f88501b642d81d3"
+            "url": "https://us.download.nvidia.com/Windows/528.24/528.24-desktop-win10-win11-64bit-international-dch-whql.exe#/dl.7z_",
```
Comment (scowalt) [inline bucket/nvidia-display-driver-np.json]: The addition of `dch` to this URL is a bug which causes us to install the wrong driver version!

I need to remove `dch` from this URL and update the hash, but _also_ change the auto-update logic to look for the new URL

File: bucket/nvidia-display-driver-dch-np.json (hunk 2 above)
Comment (scowalt) [reply bucket/nvidia-display-driver-dch-np.json]: The distinction between "DCH" and "standard" is something I found quite confusing. However, IMO most users should be installing DCH drivers

image

I added a bit of context from NVIDIA to each driver description. I've baked in a bit of pro-DCH bias, but I quoted directly from NVIDIA's descriptions

File: bucket/nvidia-display-driver-np.json (hunk 3 above)
Comment (scowalt) [reply bucket/nvidia-display-driver-np.json]: It looks like NVIDIA has stopped shipping "Standard" drivers? Maybe this is why the package broke in the first place.

Either way, I changed the auto-update logic so the "standard" package will only install the standard driver

**Comment by rasa** [comment]
```
--> Running: scoop install  nvidia-display-driver-dch-np.json

Installing 'nvidia-display-driver-dch-np' (528.24) [64bit]
528.24-desktop-win10-win11-64bit-international-dch-whql.exe (813.4 MB) [==========================================] 100%
Checking hash of 528.24-desktop-win10-win11-64bit-international-dch-whql.exe ... ok.
Running installer script...
Linking ~\scoop\apps\nvidia-display-driver-dch-np\current => ~\scoop\apps\nvidia-display-driver-dch-np\528.24
'nvidia-display-driver-dch-np' (528.24) was installed successfully!
Notes
-----
This is only an installer for the NVIDIA display driver. Running `scoop uninstall nvidia-display-driver-dch-np` will only
unregister it from Scoop.
The NVIDIA display driver installer installs and enables the NVIDIA Display Container LS service, which is required for the
NVIDIA Control Panel application but also contains telemetry components.
To disable this functionality and prevent it from being re-enabled on future updates, disable the service using the
following commands:
$ Stop-Service NVDisplay.ContainerLocalSystem
$ Set-Service NVDisplay.ContainerLocalSystem -StartupType Disabled

--> Refreshing environment variables

--> Comparing ARP entries
```

**Pull request merged**